src/nupic/math/ArrayAlgo.hpp

```diff
@@ -94,19 +99,25 @@ namespace nupic {
   #elif defined(NTA_ARCH_64)
     #if defined(NTA_OS_LINUX) || defined(NTA_OS_DARWIN)
 
-    __asm__ __volatile__ (
-                         "pushq  %%rbx\n\t"
-
-                         "movl   $1, %%eax\n\t"
-                         "cpuid\n\t"
-                         "movl   %%ecx, %0\n\t"
-                         "movl   %%edx, %1\n\t"
-
-                         "popq  %%rbx\n\t"
-                         : "=c" (c), "=d" (d)
-                         :
-                         :
-			 );
+      __asm__ __volatile__ (
+                            "pushq  %%rbx\n\t"
+
+                            "movl   $1, %%eax\n\t"
+                            "cpuid\n\t"
+                            "movl   %%ecx, %0\n\t"
+                            "movl   %%edx, %1\n\t"
+
+                            "popq  %%rbx\n\t"
+                            : "=c" (c), "=d" (d)
+                            :
+                            :
+                            );
+   #elif defined(NTA_OS_WINDOWS) && defined(NTA_COMPILER_MSVC)
+
+      std::array<int, 4> cpui;
```

Comment:
Microsoft have pulled away from inline assembler and have moved to compiler intrinsics (https://msdn.microsoft.com/en-us/library/26td21ds.aspx). This cpuid guarded section shows the use of that. To use the assembler on all platforms would require this asm to be separated out into a file and the build updated to assemble them. Or, leave this in place for almost all platforms, and either leave intrinsics in place or extract into an asm file for Windows builds.